tests/Tests/ORM/Functional/Ticket/GH10889Test.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * @see   https://github.com/doctrine/orm/issues/10889
+ *
+ * @group GH10889
+ */
+class GH10889Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->createSchemaForModels(
+            GH10889User::class,
+            GH10889Item::class,
+            GH10889Box::class
+        );
+    }
+
+    public function testIssue(): void
+    {
+        $user = new GH10889User();
+        $box  = new GH10889Box($user, null);
+
+        $this->_em->persist($user);
+        $this->_em->persist($box);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        /** @var list<GH10889Box> $boxes */
+        $boxes = $this->_em
+            ->getRepository(GH10889Box::class)
+            ->createQueryBuilder('box')
+            ->leftJoin('box.item', 'item')->addSelect('item')
+            ->getQuery()
+            ->getResult();
+
+        $this->assertArrayHasKey(0, $boxes);
+        $this->assertEquals(1, $boxes[0]->user->id);
+        $this->assertNull($boxes[0]->item);
+    }
+}
+
+/**
+ * @ORM\Entity
+ */
+class GH10889User
+{
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="integer")
+     * @ORM\GeneratedValue
+     *
+     * @var int
+     */
+    public $id;
+}
+
+/**
+ * @ORM\Entity
+ */
+class GH10889Item
+{
+    /**
+     * @ORM\Id
+     * @ORM\Column(type="integer")
+     * @ORM\GeneratedValue
+     *
+     * @var int
+     */
+    public $id;
+}
+
+/**
+ * @ORM\Entity
+ */
+class GH10889Box
+{
+    /**
+     * @ORM\Id
+     * @ORM\OneToOne(targetEntity="GH10889User")
+     *
+     * @var GH10889User
+     */
+    public $user;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="GH10889Item")
```

Comment:
Many boxes have one item?

Reply:
Yes, the same item can be put in the box of different user.
Why are you so picky for this example? It's just here to mimic a real case while only focusing on the issue. I can clutter it with ten more properties to make it an interesting piece of software but that does not serve the point in my opinion.
Or should I call them class A B and C so we can move on with the real issue I'm trying so solve here? ^^

Reply:
My hope is that if I can make sense if the test data, I'll be able to make sense of the test, and then the source code change. And if you can't be think of 2 relationships that, you know, make sense, then use A, B, C, at least it is not misleading. Anyway, a user could have a single contract with an internet provider.

Reply:
Ok I see. But my new example works fine. Each user have 1 box where they can store 1 item of their choice. Multiple users can store the same item (like item you put in a cart)

Reply:
I'll try to keep that in mind when giving this another review, but I would rather not.

Reply:
Example that I think can work:
- Person
- CurriculumVitae
- Company

A Person has 1 CV, and that CV could have a `currentCompany` field, nullable, and allowing only one company.

Reply:
@greg0ire I changed it with your example :)